Platform/RaspberryPi/Drivers/Genet/Genet.c

```diff
@@ -0,0 +1,132 @@
+/** @file
+
+  Copyright (c) 2019, Jeremy Linton All rights reserved.<BR>
+
+  SPDX-License-Identifier: BSD-2-Clause-Patent
+
+  This driver acts like a stub to set the BCM 
+  Genet MAC address, until the actual network driver
+  is in place. This should allow us to retrieve the
+  mac address directly from the hardware in supported
+  OS's rather than passing it via DSDT (which isn't 
+  ideal for a number of reasons, foremost the hardware
+  should be self describing).
+
+**/
+
+#include <Library/ArmLib.h>
+#include <Library/DebugLib.h>
+#include <Library/DevicePathLib.h>
+#include <Library/IoLib.h>
+#include <Library/UefiBootServicesTableLib.h>
+#include <Library/UefiLib.h>
+
+#include <PiDxe.h>
+#include <Protocol/RpiFirmware.h>
+
+#define GENET_BASE                 0xfd580000  // len = 0x10000
+#define GENET_SYS_RBUF_FLUSH_CTRL  0x0008
+#define GENET_UMAC_MAC0            0x080C
+#define GENET_UMAC_MAC1            0x0810
+
+STATIC
+VOID
+RMWRegister (
+  UINT32                Offset,
+  UINT32                Mask,
+  UINT32                In
+  )
+{
+  EFI_PHYSICAL_ADDRESS  Addr = GENET_BASE;
+  UINT32                Data = 0;
+  UINT32                Shift;
+
```

Comment:
Please don't use initializers. It's a stupid rule, and we already violate it in places in the RPi port, but I'd rather keep branc new code clean in this respect